### PR TITLE
ASA-334: Pre-populate form if user comes from /check_answers page

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala [hunk 1]
```diff
@@ -52,7 +53,25 @@ class AMLSController @Inject()(
...
+            case (Some(amlsDetails), mayBeGoBackUrl) =>
+              val form: Map[String, String] =
+                Map(
+                  "amlsCode"         -> amlsBodies.find(_._2 == amlsDetails.supervisoryBody).map(_._1).getOrElse(""),
+                  "membershipNumber" -> amlsDetails.membershipNumber,
+                  "expiry.day"       -> amlsDetails.membershipExpiresOn.getDayOfMonth.toString,
+                  "expiry.month"     -> amlsDetails.membershipExpiresOn.getMonthValue.toString,
+                  "expiry.year"      -> amlsDetails.membershipExpiresOn.getYear.toString
+                )
+              Ok(html.money_laundering_compliance(amlsForm(amlsBodies.keySet).bind(form), amlsBodies, mayBeGoBackUrl))
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala]: this pre-populates the amls details already stored in the cache (if there are any)

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala]: and also passing the goBack url (if stored in session already)

File: app/uk/gov/hmrc/agentsubscriptionfrontend/util/package.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend
+
+import scala.concurrent.Future
+
+package object util {
+
+  implicit def toFuture[T](a: T): Future[T] = Future.successful(a)
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/util/package.scala]: Like this - thanks! :)

File: app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala [hunk 3]
```diff
@@ -63,7 +82,7 @@ class AMLSController @Inject()(
         amlsForm(amlsBodies.keys.toSet)
           .bindFromRequest()
           .fold(
-            formWithErrors => Future.successful(Ok(html.money_laundering_compliance(formWithErrors, amlsBodies))),
+            formWithErrors => toFuture(Ok(html.money_laundering_compliance(formWithErrors, amlsBodies))),
             validForm => {
               val amlsDetails = AMLSDetails(
                 amlsBodies.getOrElse(validForm.amlsCode, throw new Exception("Invalid AMLS code")),
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala]: On submission, this looks like it redirects directly to the `showCheckAnswers` page without being subjected to the check for clean/unclean credentials (https://confluence.tools.tax.service.gov.uk/display/AG/Subscription+-+journey+and+pages). Could we use a similar routing to the one that the /business-email submission endpoint uses?

**Pull request merged**